cmd/collect-profiles/main.go

```diff
@@ -0,0 +1,339 @@
+package main
+
+import (
+	"bytes"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/tls"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/pem"
+	"fmt"
+	"io"
+	"math/big"
+	"net/http"
+	"net/url"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/spf13/cobra"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/klog/v2"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/openshift/operator-framework-olm/pkg/profiling/config"
+)
+
+const (
+	profileConfigMapLabelKey = "olm.openshift.io/pprof"
+)
+
+var (
+	rootCmd = newCmd()
+
+	// Used for flags
+	namespace    string
+	configPath   string
+	clientCAPath string
+)
+
+func init() {
+	rootCmd.PersistentFlags().StringVarP(&namespace, "namespace", "n", "default", "The Kubernetes namespace where the generated configMaps should exist. Defaults to \"default\".")
+	rootCmd.MarkFlagRequired("namespace")
+	rootCmd.PersistentFlags().StringVarP(&configPath, "config-path", "c", "/etc/config", "The path to the collect-profiles configuration file.")
+	rootCmd.MarkFlagRequired("config-file")
+	rootCmd.PersistentFlags().StringVarP(&clientCAPath, "client-ca", "", "/var/run/secrets/serving-cert", "The path to the tls cert used by the client making https requests against the pprof URLs.")
```

Comment:
It might improve clarity to rename this (maybe to just `cert`?), since it's a path to the directory containing the certificate & key and not to a CA bundle.

Reply:
Renamed the following flags:
- `config-path` to `config-mount-path`
-  `client-ca` to `cert-mount-path`